test/sanity/validate-modules/utils.py

```diff
@@ -57,8 +56,6 @@ def __enter__(self):
         self.sys_stderr = sys.stderr
         sys.stdout = self.stdout = StringIO()
         sys.stderr = self.stderr = StringIO()
-        setattr(sys.stdout, 'encoding', self.sys_stdout.encoding)
-        setattr(sys.stderr, 'encoding', self.sys_stderr.encoding)
```

Comment:
This is the hardest part but I think I finally got it.

Because StringIO and BytesIO in Python š don't support encoding, I'll use BytesIO in combination with TextIOWrapper because TextIOWrapper supports encoding specification.